### ISSUE TITLE
Please add some info on what is in this repo with respect to Sparse 

### ISSUE THREAD

**Issue by dibyendumajumdar** [lucvoo/sparse-dev#1]
Hi,

Both with this and the sparse repo - I feel unsure what is included here. It would be great if you could add some doc on what the main features/changes are in these two repos, as at present (in my case) I am unsure whether to look into the changes here or not.

Regards
Dibyendu

**Comment by lucvoo** [comment]
Hi,

For sparse-dev, it's really a development tree where I rebase & delete branches all the time.
So unless I post a series on the ML with a link here and "saying for review and/or testing",
I won't guarantee anything. But if I received some review, I'll post an update with "-v1, -v2"
appended to the name of the branch so it's still easy to compare them. The branch 'master'
is special and is the same as in the 'sparse' tree. The branch 'tip' is also special and can be considered as a candidate for master but is still rebased all the time.

When I fell that a branch is good & stable enough, I merge it in my other tree 'sparse'
which is never rebased, well tested and has no topic branches. Soon after I'll update things
here and delete the branch that was merged.

In short, everything here must be considered as ephemeral and of unknown quality. OTOH, 
the 'sparse' tree is stable & well tested. So, if you want to use some code better to take it from the other tree.

About describing the changes in the master branch, well yes, sure. I've planned to do this soon, maybe two weeks or so.

**Comment by dibyendumajumdar** [comment]
Okay thanks, that's good to know. Might be worth mentioning this somewhere. 

I will try to create a patch for my LLVM changes based on your main repo - that way I can run my tests. This will allow me to get confidence about applying your changes. So if you see me forking your repo it will be for that reason.

**Comment by lucvoo** [comment]
I've put this explanation on the wiki here: https://github.com/lucvoo/sparse-dev/wiki
